// ng-sweetalert2-slc/src/index.ts
declare var swal: any;

import { Injectable } from '@angular/core';

@Injectable()
export class SweetAlertService {
  constructor() { }

  swal(args = {}) {
    return swal(args);
  }

  prompt(options) {
    const baseOptions = {
      showCancelButton: true,
      confirmButtonText: 'Submit',
      input: 'text'
    };
    return swal(Object.assign({}, baseOptions, options));
  }

  confirm(options) {
    const baseOptions = {
      showCancelButton: true,
      confirmButtonText: 'Confirm',
      type: 'warning'
    };
    return swal(Object.assign(baseOptions, options));
  }

  alert(options) {
    const baseOptions = {
      confirmButtonText: 'OK',
      type: 'info'
    };
    return swal(Object.assign(baseOptions, options));
  }

  question(options) {
    return this.alert(Object.assign({ type: 'question' }, options));
  }

  success(options) {
    return this.alert(Object.assign({ type: 'success' }, options));
  }

  error(options) {
    return this.alert(Object.assign({ type: 'error' }, options));
  }

  warning(options) {
    return this.alert(Object.assign({ type: 'warning' }, options));
  }

  info(options) {
    return this.alert(Object.assign({ type: 'info' }, options));
  }
}
