// angular2-markdown/markdown/markdown.module.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { HttpModule } from '@angular/http';
import { MarkdownComponent } from './markdown.component';
import { MarkdownService } from './markdown.service';
import { MarkdownConfig } from './markdown.config';
var MarkdownModule = MarkdownModule_1 = (function () {
    function MarkdownModule() {
    }
    MarkdownModule.forRoot = function () {
        return {
            ngModule: MarkdownModule_1,
            providers: [MarkdownConfig]
        };
    };
    return MarkdownModule;
}());
MarkdownModule = MarkdownModule_1 = __decorate([
    NgModule({
        imports: [CommonModule, HttpModule],
        declarations: [MarkdownComponent],
        providers: [MarkdownService],
        exports: [MarkdownComponent],
        entryComponents: [MarkdownComponent]
    })
], MarkdownModule);
export { MarkdownModule };
var MarkdownModule_1;
