// ng-sweetalert2-slc/index.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
define(["require", "exports", "@angular/core"], function (require, exports, core_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var SweetAlertService = (function () {
        function SweetAlertService() {
        }
        SweetAlertService.prototype.swal = function (args) {
            if (args === void 0) { args = {}; }
            return swal(args);
        };
        SweetAlertService.prototype.prompt = function (options) {
            var baseOptions = {
                showCancelButton: true,
                confirmButtonText: 'Submit',
                input: 'text'
            };
            return swal(Object.assign({}, baseOptions, options));
        };
        SweetAlertService.prototype.confirm = function (options) {
            var baseOptions = {
                showCancelButton: true,
                confirmButtonText: 'Confirm',
                type: 'warning'
            };
            return swal(Object.assign(baseOptions, options));
        };
        SweetAlertService.prototype.alert = function (options) {
            var baseOptions = {
                confirmButtonText: 'OK',
                type: 'info'
            };
            return swal(Object.assign(baseOptions, options));
        };
        SweetAlertService.prototype.question = function (options) {
            return this.alert(Object.assign({ type: 'question' }, options));
        };
        SweetAlertService.prototype.success = function (options) {
            return this.alert(Object.assign({ type: 'success' }, options));
        };
        SweetAlertService.prototype.error = function (options) {
            return this.alert(Object.assign({ type: 'error' }, options));
        };
        SweetAlertService.prototype.warning = function (options) {
            return this.alert(Object.assign({ type: 'warning' }, options));
        };
        SweetAlertService.prototype.info = function (options) {
            return this.alert(Object.assign({ type: 'info' }, options));
        };
        return SweetAlertService;
    }());
    SweetAlertService = __decorate([
        core_1.Injectable(),
        __metadata("design:paramtypes", [])
    ], SweetAlertService);
    exports.SweetAlertService = SweetAlertService;
});
